docs/src/08_concepts.adoc

```diff
@@ -95,42 +246,64 @@ erDiagram
     
     QUESTION_TEMPLATE ||--|| QUESTION_TYPE:has
     USER ||--|| USER_HISTORY:has
+    USER ||--|| USER_PROFILE:has
 
-    USER{
-        Long id
+    USER {
         String username
         String password
+        Date createdAt
     }
 
-    USER_HISTORY{
+    USER_HISTORY {
         int passedQuestions
         int wrongQuestions
         int gamesPlayed
-        String timesPlayed
+        int timesPlayed
         int points
     }
+
+    USER_PROFILE{
+        String bio
+        String pic
+    }
+
+    QUESTION {
+        String question
+        Object[] answers
+        int correctAnswerId
+        String image_opt
+    }
     
     QUESTION_TEMPLATE{
-        Long id
-        String template
+        String questionTemplate
     }
     
     QUESTION_TYPE{
         String name
         String query
+        String[] entities
     }
 ....
 
 |===
 | Document | Description 
 | USER 
 | Holds all the data relationed to a user registered into our game. It
-has a relation with *ONE USER_HISTORY* document which represents the records/statistics 
-of them as it is a system requisite.  
+has a relation with *ONE USER_HISTORY* and *ONE USER_PROFILe* document
+| USER_HISTORY
+| Holds all the records/statistics of a single user through all played games.
+| USER_PROFILE
+| Holds specific customization of the profile of a user.
 | QUESTION_TEMPLATE
 | Holds the template needed to query the WikiData API for the Question generation. It
-has a reation with *ONE QUESTION_TYPE* document which represents a category for that
-question (e.g. Capitals) together with its SPARQL query.
+has a reation with *ONE QUESTION_TYPE* document 
+| QUESTION_TYPE
+| Represents a category for a question (e.g. Capitals) together with its SPARQL query.
+Also, several entities are provided so prior to send SPARQL query into Wikidata Service, it
+can be selected one entity to change the whole query (e.g. Capitals of Europe or Capitals of Asia)
+| QUESTION
+| Holds document of already generated questions. This schema is used for performance enhancements
+(for a simple yet powerful cache)
 |===
 
 ===== Connection to MongoDB
```

Comment:
The diagram in this part seems to imply that the userservice and the authservice communicate with the question database and the question service with the user database. I understand this is not what we are trying to represent, but to someone external it may be confusing.

Reply:
Ok that's true. I'll divide this section into two to make a clear distinction of the different databases.

Reply:
I think you misunderstood what I was trying to say. I  was refering to the Connection to MongoDB section.

Reply:
mb, now is ok?